gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,202 @@
+                  // Nessie Content.id
+                  optional(3, COL_CONTENT_ID, Types.StringType.get()),
+                  // Nessie Content.type
+                  optional(4, COL_CONTENT_TYPE, Types.StringType.get()),
+                  // Iceberg Table/View Content's snapshot/version id.
+                  optional(5, COL_SNAPSHOT_ID, Types.LongType.get()),
+                  // Name of the reference via which the contentID was collected
+                  optional(6, COL_REFERENCE_NAME, Types.StringType.get()),
+                  // Hash of the reference via which the contentID was collected
+                  optional(7, COL_HASH_ON_REFERENCE, Types.StringType.get()))
+              .fields());
+
+  private final StructType schema = SparkSchemaUtil.convert(icebergSchema);
+
+  private final SparkSession sparkSession;
+  private final String catalogAndTableWithRefName;
+
+  public IdentifiedResultsRepo(
+      SparkSession sparkSession, String catalog, String gcRefName, String gcTableIdentifier) {
+    this.sparkSession = sparkSession;
+    this.catalogAndTableWithRefName = withRefName(catalog, gcTableIdentifier, gcRefName);
+    createTableIfAbsent(sparkSession, catalog, TableIdentifier.parse(gcTableIdentifier), gcRefName);
+  }
+
+  public StructType getSchema() {
+    return schema;
+  }
+
+  /**
+   * Collect the expired contents for the given run id as spark dataset.
+   *
+   * @param runId run id of completed identify task.
+   * @return spark dataset of row where each row is having the expired contents per content id per
+   *     reference.
+   */
+  public Dataset<Row> collectExpiredContentsAsDataSet(String runId) {
+    // Example Query:
+    // SELECT * FROM nessie.db1.`identified_results@someGcRef`
+    //    WHERE gcRunId = '9e809dc9-ac35-41c9-b1c0-5ebcf452ea2d'
+    return sql(
+        "SELECT * FROM %s WHERE %s = '%s'",
+        catalogAndTableWithRefName,
+        //
+        COL_GC_RUN_ID,
+        runId);
+  }
+
+  public String getLatestCompletedRunID() {
+    // collect row for the last written run id
+    // Example query:
+    // SELECT gcRunId FROM nessie.db2.`identified_results@someGcBranch` WHERE gcRunStart =
+    //    (SELECT MAX(gcRunStart) FROM nessie.db2.`identified_results@someGcBranch`) LIMIT 1
+    List<Row> rows =
+        sql(
+                "SELECT %s FROM %s WHERE %s = (SELECT MAX(%s) FROM %s) LIMIT 1",
+                COL_GC_RUN_ID,
+                //
+                catalogAndTableWithRefName,
+                //
+                COL_GC_RUN_START,
+                COL_GC_RUN_START,
+                catalogAndTableWithRefName)
+            .collectAsList();
+    return rows.isEmpty() ? null : rows.get(0).getString(0);
+  }
+
+  void writeToOutputTable(Dataset<Row> rowDataset) {
+    try {
+      // write content rows to the output table
+      rowDataset.writeTo(catalogAndTableWithRefName).append();
+    } catch (NoSuchTableException e) {
+      throw new RuntimeException(e);
```

Comment:
can you give a descriptive message? smth like: `"Problem when writing the rowDataset: " +  rowDataset + " to " + catalogAndTableWithRefName`